libminifi/test/flow-tests/FlowControllerTests.cpp

```diff
@@ -133,12 +144,9 @@ TEST_CASE("Flow shutdown waits for a while", "[TestFlow2]") {
   testController.startFlow();
 
   // wait for the source processor to enqueue its flowFiles
-  int tryCount = 0;
-  while (tryCount++ < 10 && root->getTotalFlowFileCount() != 3) {
-    std::this_thread::sleep_for(std::chrono::milliseconds{20});
-  }
+  busy_wait(std::chrono::milliseconds{50}, [&] {return root->getTotalFlowFileCount() != 0;});
```

Comment:
Is there no way to avoid changing `== 3` to `!= 0`?  The test condition is much weaker now.

Reply:
only seemingly weaker, since the `3` flowFiles are generated in batch, and their insertion is one atomic operation

Reply:
wrote back the `== 3` as it should make no difference